samples/tutorials/mnist/02_Local_Development_with_Docker_Image_Components.ipynb

```diff
@@ -0,0 +1,548 @@
+    "%%bash\n",
+    "\n",
+    "# Create folders if they don't exist.\n",
+    "mkdir -p tmp/components/mnist_training\n",
+    "\n",
+    "# Create the Python file that lists GCS blobs.\n",
+    "cat > ./tmp/components/mnist_training/app.py <<HERE\n",
+    "import argparse\n",
+    "from datetime import datetime\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "parser = argparse.ArgumentParser()\n",
+    "parser.add_argument(\n",
+    "    '--model_file', type=str, required=True, help='Name of the model file.')\n",
+    "parser.add_argument(\n",
+    "    '--bucket', type=str, required=True, help='GCS bucket name.')\n",
+    "args = parser.parse_args()\n",
+    "\n",
+    "bucket=args.bucket\n",
+    "model_file=args.model_file\n",
+    "\n",
+    "model = tf.keras.models.Sequential([\n",
+    "  tf.keras.layers.Flatten(input_shape=(28, 28)),\n",
+    "  tf.keras.layers.Dense(512, activation=tf.nn.relu),\n",
+    "  tf.keras.layers.Dropout(0.2),\n",
+    "  tf.keras.layers.Dense(10, activation=tf.nn.softmax)\n",
+    "])\n",
+    "\n",
+    "model.compile(optimizer='adam',\n",
+    "              loss='sparse_categorical_crossentropy',\n",
+    "              metrics=['accuracy'])\n",
+    "\n",
+    "print(model.summary())    \n",
+    "\n",
+    "mnist = tf.keras.datasets.mnist\n",
+    "(x_train, y_train),(x_test, y_test) = mnist.load_data()\n",
+    "x_train, x_test = x_train / 255.0, x_test / 255.0\n",
+    "\n",
+    "callbacks = [\n",
+    "  tf.keras.callbacks.TensorBoard(log_dir=bucket + '/logs/' + datetime.now().date().__str__()),\n",
+    "  # Interrupt training if val_loss stops improving for over 2 epochs\n",
+    "  tf.keras.callbacks.EarlyStopping(patience=2, monitor='val_loss'),\n",
+    "]\n",
+    "\n",
+    "model.fit(x_train, y_train, batch_size=32, epochs=5, callbacks=callbacks,\n",
+    "          validation_data=(x_test, y_test))\n",
+    "\n",
+    "\n",
+    "model.save(model_file)\n",
+    "\n",
+    "from tensorflow import gfile\n",
+    "\n",
+    "gcs_path = bucket + \"/\" + model_file\n",
+    "\n",
+    "if gfile.Exists(gcs_path):\n",
+    "    gfile.Remove(gcs_path)\n",
+    "\n",
+    "gfile.Copy(model_file, gcs_path)\n",
+    "with open('/output.txt', 'w') as f:\n",
+    "  f.write(gcs_path)\n",
+    "HERE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create docker file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now create a container that runs the script. Start by creating a `Dockerfile`. A `Dockerfile` contains the instructions to assemble a Docker image. The `FROM` statement specifies the Base Image from which you are building. `WORKDIR` sets the working directory. When you assemble the Docker image, `COPY` will copy the required files and directories (for example, `app.py`) to the filesystem of the container. `RUN` will execute a command (for example, install the dependencies) and commits the results. "
```

Comment:
Dockerfile doesn't need to be in code font

`Dockerfile` -> Dockerfile

Reply:
will copy -> copies
filesystem -> file system
will execute -> executes

Reply:
Modified as suggested.